src/components/locations/SingleGym.jsx

```diff
@@ -20,30 +20,80 @@ const SingleGym = () => {
         px: 2,
         borderRadius: 2,
       }}>
-        <Typography variant="h3" className="centered-text" sx={{ py: 2, color: theme => theme.palette.common.white, }}>{gymInfo.name}</Typography>
-        <Container style={{ overflowY: 'scroll', height: '40rem' }} sx={{ borderRadius: 2, bgcolor: theme => theme.palette.common.white}}>
+        <Typography
+          variant="h3"
+          className="centered-text"
+          sx={{ py: 2, color: theme => theme.palette.common.white, }}
+        >
+          {gymInfo.name}
+        </Typography>
+        <Container
+          style={{ overflowY: 'scroll', height: '40rem' }}
+          sx={{ borderRadius: 2, bgcolor: theme => theme.palette.common.white}}
+          data-testid="main-container"
+        >
           <Link to={`/sections/${gymInfo.id}`}>
             Sections
           </Link>
-          <Typography variant="h6" className="centered-text">{gymInfo.address}</Typography>
-          <Typography variant="h6" className="centered-text">{gymInfo.phoneNumber}</Typography>
-          <Typography variant="h6" className="centered-text">{`Facebook: ${gymInfo.facebook !== null ? gymInfo.facebook : 'None available'}`}</Typography>
-          <Typography variant="h6" className="centered-text">{`Instagram: ${gymInfo.instagram !== null ? gymInfo.instagram : 'None available'}`}</Typography>
-          <Typography variant="h6" className="centered-text">{`Twitter: ${gymInfo.twitter !== null ? gymInfo.twitter : 'None available'}`}</Typography>
-
-          <Typography variant="h4" className="centered-text">The Setters:</Typography>
-          <Typography variant="h6" className="centered-text">Head Setter:</Typography>
-          
-          <List className="centered-text" sx={{ bgcolor: theme => theme.palette.common.white}}>
+          <Typography
```

Comment:
consider:


```
const renderTypography = (label, content) => (
  <Typography variant="h6" className="centered-text">
    {label ? `${label}: ` : ''}{content || 'None available'}
  </Typography>
);

return (
  <>
    {renderTypography(null, gymInfo.address)}
    {renderTypography(null, gymInfo.phoneNumber)}
    {renderTypography('Facebook', gymInfo.facebook)}
    {renderTypography('Instagram', gymInfo.instagram)}
    {renderTypography('Twitter', gymInfo.twitter)}
    <Typography variant="h4" className="centered-text">The Setters:</Typography>
    {renderTypography('Head Setter', null)}
  </>
);
```

May not be great but just reduces duplicaiton of Typography tag